src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -90,7 +92,9 @@ public class Session {
                 LoadUserProfile = true
             };
 
-            if (WindowsIdentity.GetCurrent().User != ((WindowsIdentity)User).User) {
+            var useridentity = User as WindowsIdentity;
+            if (useridentity != null && WindowsIdentity.GetCurrent().User != useridentity.User) {
+                logger.LogInformation($"Creating user environment variables for [{User.Name}] profile: {profilePath}");
                 // if broker and rhost are run as different users.
                 psi.EnvironmentVariables["USERNAME"] = username.ToString();
```

Comment:
Might be worth logging the computed values as well, in case something goes awry, so that we can quickly check what they were in the logs.

Also, same thing about verbosity levels as above applies throughout this method - most everything that does not report an error should be `LogTrace`, except logging that indicate start/end of important operations, e.g. "Starting host process for session {Id} of user {Name}" at the beginning, and "Host process started successfully for session {Id} of user {Name}" after `_process.Start()`.